### PR TITLE
Category Context and Reducer, Product Context And Reducer added

### PR DESCRIPTION
**Pull request by Mahi24-prog**
### Following Functionality Added
- Showing Products depend on category selected on home page
- All filters added 
- We can filters product by Price, Price high to low and low to high, Rating, Delivery, Out of stock, Clear

https://user-images.githubusercontent.com/59019169/161368667-08707b8e-021b-411c-a76e-c2010ba2c18f.mp4

### CODE REVIEW

**Review by sakleprajwal — APPROVED** [review]
Everything looks good please go ahead

File: src/Components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -5,7 +5,8 @@ const Navbar = () => {
     return (
         <>
             <nav id="home-navbar" className="flex-ali-cen flex-wrap">
-        <div className="home-nav-left">
+        <div className="home-nav-left d-flex">
+            <img src='images/brand-logo.png' className='brand-logo'/>
```
Comment (prathmeshjagtap) [inline src/Components/Navbar/Navbar.jsx:9]: add an alt tag to image

File: src/Contexts/ProductContext.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const [productsData, setProductsData] = useState([])
+  const [state, dispatch] = useReducer(ProductReducer, { productsCategoryData: [],sortBy: null, starRating: "1_And_Above",
+      priceFilterValue : 1000000, outOfStock: true, fastDelivery : false
+  })
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await fetch("/api/products", { method: "GET" })
+        console.log(response)
```
Comment (prathmeshjagtap) [inline src/Contexts/ProductContext.jsx:18]: remove console.log

File: src/Reducer/ProductReducer.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+export function ProductReducer(state, action) {
+    switch (action.type) {
+        case "SORT": return ({ ...state, sortBy: action.payload })
+        case "FilterRating" : return({...state, starRating: action.payload})
+        case "CategoryFilter" : 
+            console.log(action.payload)
```
Comment (prathmeshjagtap) [inline src/Reducer/ProductReducer.jsx:6]: remove console.log

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]
Else everything is good

File: src/Contexts/ProductContext.jsx (hunk 2 above)
Comment (Mahi24-prog) [reply src/Contexts/ProductContext.jsx:18]: ok

**Review by prathmeshjagtap — APPROVED** [review]

File: src/Reducer/ProductReducer.jsx (hunk 3 above)
Comment (Mahi24-prog) [reply src/Reducer/ProductReducer.jsx:6]: ok

**Pull request merged**